pkg/azure/actuator.go

```diff
@@ -119,6 +119,40 @@ func isAzureCredentials(providerSpec *runtime.RawExtension) (bool, error) {
 	return isAzure, nil
 }
 
+// needsUpdate will return whether the current credentials satisfy what's being requested
+// in the CredentialsRequest
+func (a *Actuator) needsUpdate(ctx context.Context, cr *minterv1.CredentialsRequest) (bool, error) {
+	logger := a.getLogger(cr)
+	// If the secret simply doesn't exist, we definitely need an update
+	exists, err := a.Exists(ctx, cr)
+	if err != nil {
+		return true, err
+	}
+	if !exists {
+		return true, nil
+	}
+
+	azureStatus, err := decodeProviderStatus(a.codec, cr)
+	if err != nil {
+		return true, fmt.Errorf("unable to decode ProviderStatus: %v", err)
+	}
+
+	if azureStatus.ServicePrincipalName == "" {
+		// passthrough-specifc checks here
+
+		credentialsRootSecret, err := a.GetCredentialsRootSecret(ctx, cr)
+		if err != nil {
+			log.WithError(err).Debug("error retrieving cloud credentials secret")
```

Comment:
return the error here as it makes no sense to be in passthrough mode and have the root cloud creds be missing